ocs_ci/helpers/dr_helpers.py

```diff
@@ -73,3 +76,177 @@ def relocate(preferred_cluster, drpc_name, namespace):
+        config.switch_ctx(cluster.MULTICLUSTER["multicluster_index"])
+        logger.info(
+            f"Validating mirroring status on cluster {cluster.ENV_DATA['cluster_name']}"
+        )
+        sample = TimeoutSampler(
+            timeout=timeout,
+            sleep=5,
+            func=check_mirroring_status_ok,
+            replaying_images=replaying_images,
+        )
+        assert sample.wait_for_func_status(result=True), (
+            "The mirroring status does not have expected values within the time"
+            f" limit on cluster {cluster.ENV_DATA['cluster_name']}"
+        )
+
+
+def get_all_vrs(namespace=None):
+    """
+    Gets all VRs in given namespace
+
+    Args:
+        namespace (str): the namespace of the VR resources
+
+    Returns:
+         list: list of all VR in namespace
+
+    """
+    vr_obj = ocp.OCP(kind=constants.VOLUME_REPLICATION, namespace=namespace)
+    vr_items = vr_obj.get().get("items")
+    vr_list = [vr.get("metadata").get("name") for vr in vr_items]
+    return vr_list
+
+
+def check_vr_status(state, namespace):
+    """
+    Check if all VR in the given namespace are in expected state
+
+    Args:
+        state (str): The VR state to check for (e.g. 'primary', 'secondary')
+        namespace (str): the namespace of the VR resources
+
+    Returns:
+        bool: True if all VR are in expected state, False otherwise
+
+    """
+    vr_obj = ocp.OCP(kind=constants.VOLUME_REPLICATION, namespace=namespace)
+    vr_list = get_all_vrs(namespace)
+
+    vr_state_mismatch = []
+    for vr in vr_list:
+        desired_state = vr_obj.get(vr).get("spec").get("replicationState")
+        current_state = vr_obj.get(vr).get("status").get("state")
+        logger.info(
+            f"VR: {vr} desired state is {desired_state}, current state is {current_state}"
+        )
+
+        if not (
+            state.lower() == desired_state.lower()
+            and state.lower() == current_state.lower()
+        ):
+            vr_state_mismatch.append(vr)
+
+    if not vr_state_mismatch:
+        logger.info(f"All {len(vr_list)} VR are in expected state {state}")
+        return True
+    else:
+        logger.warning(
+            f"Following {len(vr_state_mismatch)} VR are not in expected state: {vr_state_mismatch}"
+        )
+        return False
+
+
+def wait_for_vr(count, namespace, state="primary", timeout=300):
+    """
+    Wait for all VR resources to exist in expected state in the given namespace
+
+    Args:
+        count (int): Expected number of VR resources
+        namespace (str): the namespace of the VR resources
+        state (str): The VR state to check for (e.g. 'primary', 'secondary')
+        timeout (int): time in seconds to wait for VR resources to be created
+            or reach expected state
+
+    Returns:
+        bool: True if all VR are in expected state
+
+    Raises:
+        Exception: In case of unexpected VR resource count or status
```

Comment:
@sidhant-agrawal , can you specify exact exceptions raise!

Reply:
Done